docs/pages/en-us/Architecture.md

```diff
@@ -100,6 +99,8 @@ When the job tracking worker receives a success message to be proxied (that is,
 
 The job tracking worker recognizes failure and retry messages, which are being proxied, and updates the job database accordingly.
 
+The job tracking worker can also automatically forward on dependent jobs for execution.  A dependent job is a job which must wait until a specific job or list of jobs have completed before it, itself can be executed.  As the job tracking worker monitors a job's progress, should a job complete, the job tracking worker will receive a list of jobs which can now be executed as the job(s) which these jobs were dependent upon have now Completed.
```

Comment:
> specific job or list of jobs have completed before it, itself can be executed.

I don't think you need the word itself or the comma here:

> specific job or list of jobs have completed before it can be executed.

Reply:
> As the job tracking worker monitors a job's progress, should a job complete, the job tracking worker will receive a list of jobs which can now be executed as the job(s) which these jobs were dependent upon have now Completed.

I would rewrite this as:

> The job tracking worker monitors a job's progress, when a job completes, the job tracking worker will receive a list of jobs which can now be executed as the job(s) which these jobs were dependent upon have now completed.